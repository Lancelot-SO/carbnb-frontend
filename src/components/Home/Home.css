.home-container {
  width: 80%;
  padding: 2rem;
  max-height: 80vh;
  min-height: 0;
  min-width: 0;
}

.home-container-inner {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding-top: 0.25rem;
}

.home-title-container {
  display: flex;
  flex-direction: column;
  justify-items: center;
  justify-self: center;
  align-items: center;
  text-align: center;
}

.home-title {
  padding-bottom: 2rem;
  font-size: 20px;
  font-weight: bold;
  text-align: center;
}

.home-description {
  padding-bottom: 0.75rem;
  padding-top: 0.5rem;
  font-size: 16px;
  color: var(--color-primary);
}

.swiper {
  margin-left: auto;
  margin-right: auto;
  position: relative;
  overflow: hidden;
  list-style: none;
  padding: 0;
    padding-right: 0px;
    padding-left: 0px;
  z-index: 1;
  display: block;
}

.swiper {
  width: 100%;
  height: 100%;
  margin: 20px auto;
}

.swiper-wrapper {
  position: relative;
  width: 100%;
  height: 100%;
  z-index: 1;
  display: flex;
  transition-property: transform;
  box-sizing: content-box;
}

.swiper-slide {
  text-align: center;
  font-size: 18px;
  background: #fff;
  border: none;
  border-radius: 10px;
  padding: 10px 0;
  width: auto;
  flex-shrink: 0;
  height: 100%;
  max-height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.swiper-wrapper {
  max-height: 100%;
  height: 100%;
  display: flex;
}

.swiper-pagination {
  padding: 10px 0;
}

.swiper-slide-content {
  border-style: none;
 border-radius: 0.75rem;
}

.append-buttons {
  text-align: center;
  margin-top: 20px;
}

.append-buttons button {
  display: inline-block;
  cursor: pointer;
  color: var(--color-secondary);
  text-decoration: none;
  padding: 4px 10px;
  border-radius: 4px;
  margin: 0 10px;
  font-size: 13px;
}

.swiper-slide {
  text-align: center;
  font-size: 18px;
  background: #fff;
  border: none;
  border-radius: 10px;
  padding: 10px 0;
  width: auto;
  flex-shrink: 0;
  height: 100%;
  max-height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.swiper-slide-img {
  display: block;
  width: auto;
  height: 40vh;
  object-fit: cover;
  border-radius: 1rem;
  width: 100%;
}

.car-swiper-slide-text > div {
  font-size: 18px;
}

.car-name {
  font-size: 22px;
  padding-top: 0.75rem;
  font-weight: 700;
}

.car-model,
.car-description {
  font-weight: 700;
  font-size: 0.875rem;
  color: var(--color-primary);
}

.car-model {
  padding-top: 0.75rem;
}

.car-description {
  padding-bottom: 0.75rem;
}

.car-bottom-links {
  position: relative;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  gap: 1.25rem;
  margin-bottom: 2.5rem;
  color: var(--color-primary);
}

.car-bottom-links > li {
  font-size: 18px;
}

@media only screen and (max-width: 874px) {
  .home-container-outer {
    width: 100%;
  }
}