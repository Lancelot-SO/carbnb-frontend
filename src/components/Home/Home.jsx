import React, { useEffect } from 'react';
import SwiperCore, { Virtual, Navigation } from 'swiper';
import { Swiper, SwiperSlide } from 'swiper/react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { fetchcars } from '../../redux/Actions/car-actions';
import 'swiper/css';
import 'swiper/css/navigation';
import 'swiper/css/pagination';
import 'swiper/css/scrollbar';

SwiperCore.use([Virtual, Navigation]);

const Home = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchcars());
  }, [dispatch]);

  const cars = useSelector((state) => state.cars);

  const breakpoints = {
    320: {
      slidesPerView: 1,
      spaceBetween: 10,
    },
    640: {
      slidesPerView: 1,
      spaceBetween: 20,
    },
    1124: {
      slidesPerView: 3,
      spaceBetween: 25,
    },
  };
  return (
    <>
    <div className="home-container-outer">
      <div className="home-container-inner">
      <div className="home-title-container">
        <h2 className="home-title">
          CarBnB 
        </h2>
        <p className="home-description">
          Please select a CarBnB Model
        </p>
      </div>
      <Swiper
        className="swiper"
        spaceBetween={10}
        slidesPerView={2}
        navigation
        breakpoints={breakpoints}
      >
        {cars.map((slideContent, index) => (
          <SwiperSlide key={slideContent.id}>
            <Link to={`/motorcycles/${slideContent.id}`} state={slideContent}>
              <div className="border-none swiper-slide-content rounded-xl">
                <img
                  src={slideContent.image}
                  className=""
                  alt={`Slide ${index + 1}, ${slideContent.name}`}
                />
                <div className="car-swiper-slide-text">
                  <div>
                    <h1 className="car-name">{slideContent.name}</h1>
                    <h4 className="car-model">{slideContent.model}</h4>
                    <p className="car-description">
                      {slideContent.description}
                    </p>
                  </div>
                  <ul className="car-bottom-links">
                    <li>
                      <a href="/">
                        <i className="fa-brands fa-facebook" />
                      </a>
                    </li>
                    <li>
                      <a href="/">
                        <i className="fa-brands fa-linkedin" />
                      </a>
                    </li>
                    <li>
                      <a href="/">
                        <i className="fa-brands fa-twitter" />
                      </a>
                    </li>
                    <li>
                      <a href="/">
                        <i className="fa-brands fa-amazon" />
                      </a>
                    </li>
                    <li>
                      <a href="/">
                        <i className="fa-brands fa-google" />
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
            </Link>
          </SwiperSlide>
        ))}
      </Swiper>
    </div>
    </div>
    </>
  );
}

export default Home;