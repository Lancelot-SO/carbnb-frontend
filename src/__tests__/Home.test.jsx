import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { describe, it, expect } from 'vitest';
import '@testing-library/jest-dom';
import store from '../redux/index';
import Home from '../components/Home/Home';

describe('App', () => {
  it('renders Home', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    );

    expect(tree).toMatchSnapshot();
  });
});
