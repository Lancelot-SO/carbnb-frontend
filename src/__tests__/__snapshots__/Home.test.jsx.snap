// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders Home 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home-container-outer"
      >
        <div
          class="home-container-inner"
        >
          <div
            class="home-title-container"
          >
            <h2
              class="home-title"
            >
              CarBnB
            </h2>
            <p
              class="home-description"
            >
              Please select a CarBnB Model
            </p>
          </div>
          <div
            class="swiper swiper-initialized swiper-horizontal"
          >
            <div
              class="swiper-wrapper"
              style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
            />
            <div
              class="swiper-button-prev"
            />
            <div
              class="swiper-button-next"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home-container-outer"
    >
      <div
        class="home-container-inner"
      >
        <div
          class="home-title-container"
        >
          <h2
            class="home-title"
          >
            CarBnB
          </h2>
          <p
            class="home-description"
          >
            Please select a CarBnB Model
          </p>
        </div>
        <div
          class="swiper swiper-initialized swiper-horizontal"
        >
          <div
            class="swiper-wrapper"
            style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
          />
          <div
            class="swiper-button-prev"
          />
          <div
            class="swiper-button-next"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
